sh/posix.bzl

```diff
@@ -347,14 +347,72 @@ _sh_posix_config = repository_rule(
 )
 
 def sh_posix_configure(name = "local_posix_config", register = True):
-    """Autodetect local Unix commands.
+    """## Usage
```

Comment:
It's a bit unusual to have function or rule docstrings start with a heading like that. 
rules_nixpkgs generates such headings by placing them into the module docstring, e.g. [here](https://github.com/tweag/rules_nixpkgs/blob/fec998bbe956ea4c40c19234a258b6bd06be605c/nixpkgs/nixpkgs.bzl#L39). Could the same work here, too?

Relatedly, rules_nixpkgs uses templates that also generate parameter lists and the like, e.g. [here](https://github.com/tweag/rules_nixpkgs/blob/fec998bbe956ea4c40c19234a258b6bd06be605c/core/docs/templates/func.vm#L12). That would be a nice improvement over the current README of rules_sh.

That said, I'm happy to leave these for a follow-up PR and already merge this one as is. After all this one achieves the goal of generating the current README from Startdoc.

Reply:
Sounds good - I'll look into this tomorrow if I have time. I've created #33 to track it.